### PR TITLE
new resource: `azapi_data_plane_resource`

### PR DESCRIPTION
**Pull request by ms-henglu**

### CODE REVIEW

File: docs/resources/azapi_data_plane_resource.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  ]
+}
+
+resource "azurerm_storage_container" "example" {
+  name                  = "content"
+  storage_account_name  = azurerm_storage_account.example.name
+  container_access_type = "private"
+}
+
+resource "azapi_data_plane_resource" "dataset" {
```
Comment (magodo) [inline docs/resources/azapi_data_plane_resource.md:101]: Is this example a bit too verbose? Can we choose another simpler one?

File: docs/resources/azapi_data_plane_resource.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+  type      = "Microsoft.Synapse/workspaces/datasets@2020-12-01"
+  parent_id = replace(azurerm_synapse_workspace.example.connectivity_endpoints.dev, "https://", "")
```
Comment (magodo) [inline docs/resources/azapi_data_plane_resource.md:103]:
```suggestion
  parent_id = trimprefix(azurerm_synapse_workspace.example.connectivity_endpoints.dev, "https://")
```

File: internal/services/azapi_data_plane_resource.go [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+func ResourceAzApiDataPlaneResource() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAzApiDataPlaneResourceCreateUpdate,
+		Read:   resourceAzApiDataPlaneResourceRead,
+		Update: resourceAzApiDataPlaneResourceCreateUpdate,
+		Delete: resourceAzApiDataPlaneResourceDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				return nil, fmt.Errorf("importing is not supported for this resource yet")
```
Comment (magodo) [inline internal/services/azapi_data_plane_resource.go:29]: Do we have a specific reason to not support importing?

**Review by magodo — COMMENTED** [review]
Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

File: internal/services/azapi_data_plane_resource.go (hunk 3 above)
Comment (ms-henglu) [reply internal/services/azapi_data_plane_resource.go:29]: It's complicated to pull the data plane resource type from the ID, I'll implement this feature in another PR/release.

**Comment by ms-henglu** [comment]
Hi @magodo ,

Thank you so much for the code review, I've updated this PR as suggested. Please take another look, thanks!

**Review by magodo — APPROVED** [review]

**Pull request merged**